ostd/libs/linux-bzimage/builder/src/lib.rs

```diff
@@ -61,7 +63,7 @@ pub fn make_bzimage(
         .unwrap()
         .read_to_end(&mut kernel)
         .unwrap();
-    let payload = kernel;
+    let payload = compress_kernel(&kernel);
```

Comment:
Currently we are only planning to support self-decompression on EFI handover 64-bit boot protocol. So compress only when `image_type` is `BzImageType::Efi64`.

This is why Linux legacy boot protocol test fails.